docs/scos/dev/guidelines/keeping-a-project-upgradable/upgradability-guidelines/private-api-is-used-on-the-project-level.md

```diff
@@ -536,14 +647,66 @@ class CustomerAccessEntityManager extends SprykerCustomerAccessEntityManager imp
 }
 ```
 
-Example of related error in the Evaluator output:
+### Resolving the error
 
-```bash
-------------------------------------------------------------------------------------------------------------------------
-Please avoid usage of PrivateApi method SprykerCustomerAccessEntityManager::getCustomerAccessEntityByContentType()
-------------------------------------------------------------------------------------------------------------------------
+To resolve the error provided in the example, try the following in the provided order:
+1. Recommended: Extend the functionality using the [Configuration strategy](/docs/scos/dev/back-end-development/extending-spryker/development-strategies/development-strategies.html#configuration).
+2. Recommended: Extend the functionality using the [Plug and Play strategy](/docs/scos/dev/back-end-development/extending-spryker/development-strategies/development-strategies.html#plug-and-play).
+3. Recommended: Extend the functionality using the [Project Modules strategy](/docs/scos/dev/back-end-development/extending-spryker/development-strategies/development-strategies.html#project-modules).
+4. Not recommended: Copy the functionality from the core to the project level:
+    1. Register the missing extension point in [Spryker Ideas](https://spryker.ideas.aha.io/), so we add it in the future.
+    2. Copy private API core entities to the project level and give them unique names. For an example, see [Example of resolving the error by copying and renaming Private API entities](#example-of-resolving-the-error-by-copying-and-renaming-private-api-entities).
+    3. As soon as the extension point in core is released, refactor the code added in step 4.2 using the strategies in steps 1-3.
+       While it's not refactored, auto-upgrades are not supported, and the effort to update the project may be bigger and require more manual work.
+
+### Example of resolving the error by copying and renaming Private API entities
+
+1. Create a new method for a factory on the project level.
```

Comment:
Are we just copying the method from the project level and renaming it?

Reply:
yes, this is how it works at the moment :)